### PR TITLE
task to delete old images

### PR DESCRIPTION
**Pull request by snopoke**
@dimagi/scale-team

### CODE REVIEW

File: custom/icds/tasks.py [hunk 1]
```diff
@@ -159,3 +165,37 @@ def run_user_indicators(phased_rollout=True):
...
+
+    start = datetime.utcnow()
+    max_age = start - timedelta(days=90)
+    db = get_blob_db()
+    paths = []
+    deleted_attachments = []
+
+    def _get_query(db_name, max_age):
+        return XFormAttachmentSQL.objects.using(db_name).filter(
+            content_type='image/jpeg',
```
Comment (czue) [inline custom/icds/tasks.py]: are all images jpegs?

**Review by czue — COMMENTED** [review]
not "approving" just cause I don't really know how the blobdb works but looks good.

File: custom/icds/tasks.py (hunk 1 above)
Comment (snopoke) [reply custom/icds/tasks.py]: For ICDS they are

File: custom/icds/tasks.py [hunk 2]
```diff
@@ -159,3 +165,37 @@ def run_user_indicators(phased_rollout=True):
             if is_monday:
                 run_indicator.delay(domain, user_id, LSSubmissionPerformanceIndicator, language_code)
                 run_indicator.delay(domain, user_id, LSVHNDSurveyIndicator, language_code)
+
+
+@periodic_task(run_every=crontab(minute=0, hour='22'))
+def delete_old_images():
+    if settings.SERVER_ENVIRONMENT not in settings.ICDS_ENVS:
+        return
```
Comment (dannyroberts) [inline custom/icds/tasks.py]: Is it possible to _only define_ this periodic task if `settings.SERVER_ENVIRONMENT not in settings.ICDS_ENVS`?

Not a huge deal, but I think that would prevent a no-op task from being enqueued once a day on other environments, which is much more of a clutter/principle issue than a performance one.

I'm thinking either

```
if settings.SERVER_ENVIRONMENT not in settings.ICDS_ENVS:
    @periodic_task(run_every=crontab(minute=0, hour='22'))
    def delete_old_images():
...
```

or

```
def delete_old_images():
...
if settings.SERVER_ENVIRONMENT not in settings.ICDS_ENVS:
    delete_old_images = periodic_task(run_every=crontab(minute=0, hour='22'))(delete_old_images)
```

File: custom/icds/tasks.py [hunk 3]
```diff
@@ -159,3 +165,37 @@ def run_user_indicators(phased_rollout=True):
...
+        return XFormAttachmentSQL.objects.using(db_name).filter(
+            content_type='image/jpeg',
+            form__domain='icds-cas',
+            form__received_on__lt=max_age
+        )
+
+    for db_name in get_db_aliases_for_partitioned_query():
+        attachments = _get_query(db_name, max_age)
+        while attachments.exists():
+            for attachment in attachments[10000]:
```
Comment (millerdev) [inline custom/icds/tasks.py]: Was `attachments[10000]` meant to be a slice (limit)? Something like `attachments[:10000]` As is, it appears to reference the 10,000th attachment, and I don't understand why you'd only want to that here.

**Comment by snopoke** [comment]
Updated

**Review by millerdev — APPROVED** [review]

**Pull request merged**